packages/launchpad/solana/client.ts

```diff
@@ -0,0 +1,483 @@
+    extParams: IInteractSolanaExt,
+  ): Promise<{
+    ixs: TransactionInstruction[];
+  }> {
+    const { launchpadProgram } = this.programs;
+
+    const authority = extParams.invoker.publicKey;
+    invariant(authority, "Undefined authority publicKey");
+
+    return {
+      ixs: [
+        await launchpadProgram.methods
+          .claimDeposits()
+          .accounts({
+            authority,
+            launchpad: launchpadKey,
+            tokenProgram: tokenProgramId,
+          })
+          .instruction(),
+      ],
+    };
+  }
+
+  async claimAllocatedVested(data: IClaimAllocatedVested, extParams: IInteractSolanaExt): Promise<ITransactionResult> {
+    if (!extParams.computeLimit) {
+      extParams.computeLimit = 280_000;
+    }
+    const { ixs, streamKeypair } = await this.prepareClaimAllocatedVestedInstructions(data, extParams);
+    const { signature } = await this.execute(ixs, extParams, streamKeypair);
+    return {
+      ixs,
+      txId: signature,
+    };
+  }
+
+  async prepareClaimAllocatedVestedInstructions(
+    { launchpad: launchpadKey, baseMint: baseMintKey, owner, tokenProgramId = TOKEN_PROGRAM_ID }: IClaimAllocatedVested,
+    extParams: IInteractSolanaExt,
+  ): Promise<{
+    ixs: TransactionInstruction[];
+    streamKeypair: Keypair;
+  }> {
+    const { launchpadProgram } = this.programs;
+
+    const payer = extParams.invoker.publicKey;
+    invariant(payer, "Undefined payer publicKey");
+
+    owner = pk(owner ?? payer);
+    if (!baseMintKey) {
+      baseMintKey = (await this.getLaunchpad(launchpadKey)).baseMint;
+    }
+    const streamKeypair = Keypair.generate();
+    const streamKey = streamKeypair.publicKey;
+    const proxyMetadataKey = deriveContractPDA(pk(this.dynamicVestingId), streamKeypair.publicKey);
+    const proxyTokensKey = getAssociatedTokenAddressSync(pk(baseMintKey), proxyMetadataKey, true);
+    const escrowKey = deriveEscrowPDA(pk(this.vestingId), streamKeypair.publicKey);
+    const depositKey = deriveDepositPDA(launchpadProgram.programId, pk(launchpadKey), owner);
+
+    const ix = await launchpadProgram.methods
+      .claimAllocatedVested()
+      .accounts({
+        depositAccount: depositKey,
+        payer,
+        proxyMetadata: proxyMetadataKey,
+        proxyTokens: proxyTokensKey,
+        streamMetadata: streamKey,
+        escrowTokens: escrowKey,
+        withdrawor: streamConstants.WITHDRAWOR_PUBLIC_KEY,
+        feeOracle: streamConstants.FEE_ORACLE_PUBLIC_KEY,
+        tokenProgram: tokenProgramId,
+      })
+      .accountsPartial({
+        streamflowProgram: this.vestingId,
+        proxyProgram: this.dynamicVestingId,
+      })
+      .instruction();
+
+    return { ixs: [ix], streamKeypair };
+  }
+
+  private async execute(
+    ixs: TransactionInstruction[],
+    extParams: IInteractSolanaExt,
+    ...partialSigners: (Keypair | undefined)[]
+  ) {
+    ixs = [...prepareBaseInstructions(this.connection, extParams), ...ixs];
```

Comment:
oh that reminds me - staking sdk doesn't have CU instructions 🤦 why did I think that prepareTransaction does it (or another wrapper..?🤔)